charts/node/templates/_helpers.tpl

```diff
@@ -98,5 +98,5 @@ Create the database path depending on the database backend in use (rocksdb or pa
 Define a regex matcher to check if the passed node flags are managed by the chart already
 */}}
 {{- define "node.chartManagedFlagsRegex" -}}
-{{- "(\\W|^)(--name|--base-path|--chain|--validator|--collator|--light|--database|--pruning|--prometheus-external|--prometheus-port|--node-key|--wasm-runtime-overrides|--jaeger-agent|--rpc-external|--unsafe-rpc-external|--ws-external|--unsafe-ws-external|--rpc-methods|--rpc-cors|--rpc-port|--ws-port|--enable-offchain-indexing)(\\W|$)" }}
+{{- "(\\W|^)(--name|--base-path|--chain|--validator|--collator|--light|--database|--pruning|--blocks-pruning|--prometheus-external|--prometheus-port|--node-key|--wasm-runtime-overrides|--jaeger-agent|--rpc-external|--unsafe-rpc-external|--ws-external|--unsafe-ws-external|--rpc-methods|--rpc-cors|--rpc-port|--ws-port|--enable-offchain-indexing)(\\W|$)" }}
```

Comment:
If we need to deploy an old node or 3rd party node which do not support `--blocks-pruning` how we can do it with a new helm chart?

Reply:
There's two ways that I can think of on how to solve this:

1. We should start using the proper appVersion value in the Chart. This approach allows us to determine which version the chart supports, and create certain conditions based from the appVersion. If we add this, it provides a visibility in which app version supports for specific version of the chart, the documentation and communication is clear just by using the chart, and it will be consistent through out different environment.

2. Another option is to just rely on specific node.image.tag condition for this. 

I prefer the first approach since it is how Helm chart should be written.

Reply:
1. appVersion will work only for parity-managed clients, but you also have kagome, moonbase which has completely different versioning.

2. image tag can be anything e.g master

One of the solutions that I see is to allow users to set `--pruning` as custom flags, for example:
```
.Values.node.chainData.pruning = -1
 .Values.node.flags = ['--pruning=1000']

```
 this will do the trick, but currently both `--pruning, --blocks-pruning` is forbidden by `chartManagedFlagsRegex`.

Reply:
Yeah that make sense. I forgot about other app implementation that uses polkadot-sdk. The solution is to allow passing of the pruning flags, but to also implement a check to disable the default pruning value if we pass the flag. This is to avoid modifying the pruning values in chart.